pkg/capabilities/consensus/ocr3/aggregators/reduce_test.go

```diff
@@ -0,0 +1,928 @@
+package aggregators_test
+
+import (
+	"math/big"
+	"testing"
+	"time"
+
+	"github.com/shopspring/decimal"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/protobuf/proto"
+
+	"github.com/smartcontractkit/libocr/commontypes"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities/consensus/ocr3/aggregators"
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities/consensus/ocr3/types"
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities/datastreams"
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink-common/pkg/values"
+	"github.com/smartcontractkit/chainlink-common/pkg/values/pb"
+)
+
+var (
+	feedIDA  = datastreams.FeedID("0x0001013ebd4ed3f5889fb5a8a52b42675c60c1a8c42bc79eaa72dcd922ac4292")
+	idABytes = feedIDA.Bytes()
+	feedIDB  = datastreams.FeedID("0x0003c317fec7fad514c67aacc6366bf2f007ce37100e3cddcacd0ccaa1f3746d")
+	idBBytes = feedIDB.Bytes()
+	now      = time.Now()
+)
+
+func TestReduceAggregator_Aggregate(t *testing.T) {
+	t.Run("happy path", func(t *testing.T) {
+		cases := []struct {
+			name                string
+			fields              []aggregators.AggregationField
+			extraConfig         map[string]any
+			observationsFactory func() map[commontypes.OracleID][]values.Value
+			shouldReport        bool
+			expectedState       any
+			expectedOutcome     map[string]any
+		}{
+			{
+				name: "empty inputkey and outputkey",
+				fields: []aggregators.AggregationField{
+					{
+						Method: "median",
+					},
+				},
+				extraConfig: map[string]any{},
+				observationsFactory: func() map[commontypes.OracleID][]values.Value {
+					mockValue, err := values.Wrap(int64(100))
+					require.NoError(t, err)
+					return map[commontypes.OracleID][]values.Value{1: {mockValue}, 2: {mockValue}, 3: {mockValue}}
+				},
+				shouldReport: true,
+				expectedOutcome: map[string]any{
+					"Reports": []any{
+						map[string]any{
+							"": int64(100),
+						},
```

Comment:
why is this a valid report, we don't want to error instead?

Reply:
Yes, I think you are right about this being an invalid report. It shouldn't be a map with an empty key, rather just a non-complex value.